pelican/paginator.py

```diff
@@ -67,8 +67,11 @@ def _get_page_range(self):
 
 class Page(object):
     def __init__(self, name, url, object_list, number, paginator, settings):
+        self.full_name = name
         self.name, self.extension = os.path.splitext(name)
-        self.base_url = url
+        dirname, filename = os.path.split(name)
+        self.base_name = dirname if filename == 'index.html' else self.name
+        self.full_url = url
```

Comment:
Someone might be using `base_url` in a template. No need to break this with no reason.

Reply:
If I parse this method (`_from_settings`) correctly, it's the values in `context` that get inserted into the patterns. I renamed the attribute from `base_url` to `full_url`, but I didn't rename the corresponding value in `context`, which is still `url`. So this shouldn't break anything, right?

Reply:
`Page` object gets passed to the template. Someone might be using `articles_page.base_url` in their templates... for something...

Reply:
Ah ok! I think the `base_` part is misleading, but I guess we're stuck with it. I've reverted to `base_url`.

Reply:
Joys of backward compatibility :)